### PR TITLE
[usdAbc] Only deliver interesting timeSamples for constant properties.

### PR DESCRIPTION
**Pull request by marsupial**
### Description of Change(s)
Only deliver the number of _unique_ samples in an Alembic archive.
### Fixes Issue(s)
By delivering all the samples, there's a larger memory foot-print on larger scenes.
More importantly, those samples get pushed to Hydra as changes:
  **pxr/usd/plugin/usdAbc/testenv/testUsdAbcIsConstant/testUsdAbcIsConstant.abc**

### CODE REVIEW

**Comment by jilliene** [comment]
Filed as internal issue #USD-5866

File: pxr/usd/plugin/usdAbc/alembicReader.cpp [hunk 1]
```diff
@@ -1689,6 +1689,16 @@ _ReaderContext::_HasValue(
...
+template <typename T> size_t
+getNumSamples(const T& object) {
+    size_t nSamples = object.getNumSamples();
+    if (!object.isConstant())
+        return nSamples;
+
+    return std::min(nSamples, size_t(1));
+}
+
 /// Fill sample times from an object with getTimeSampling() and
```
Comment (spiffmon) [inline pxr/usd/plugin/usdAbc/alembicReader.cpp]: I am not a super-expert in this code, but I don't understand why you do not need to apply the same treatment to _GetSampleTimes() ?

File: pxr/usd/plugin/usdAbc/testenv/testUsdAbcIsConstant.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+class TestUsdAbcIsConstant(unittest.TestCase):
+    def test_Read(self):
+      testFile = 'testUsdAbcIsConstant.abc'
+      stage = Usd.Stage.Open(testFile)
+      self.assertTrue(stage)
+
+      animPoints = UsdGeom.Mesh.Get(stage, "/sphere_object1")
+      self.assertTrue(animPoints)
+      self.assertEqual(animPoints.GetPrim().GetProperty('points').GetNumTimeSamples(), 240)
```
Comment (spiffmon) [inline pxr/usd/plugin/usdAbc/testenv/testUsdAbcIsConstant.py]: `UsdGeom.Tokens.points`

File: pxr/usd/plugin/usdAbc/alembicReader.cpp (hunk 1 above)
Comment (marsupial) [reply pxr/usd/plugin/usdAbc/alembicReader.cpp]: A bit confused, **_GetSampleTimes** will call though to **getNumSamples**.
In the case where **object.isConstant()** is true, it doesn't matter which sample you query as they'll all be valid (and the same), I just choose the lowest one.

Comment (spiffmon) [reply pxr/usd/plugin/usdAbc/alembicReader.cpp]: Ah, so sorry, @marsupial - I let my offline editor confuse me, and thought there was a different implementation of _GetSampleTimes() that you hadn't updated.  I was wrong!

File: pxr/usd/plugin/usdAbc/testenv/testUsdAbcIsConstant.py (hunk 2 above)
Comment (spiffmon) [reply pxr/usd/plugin/usdAbc/testenv/testUsdAbcIsConstant.py]: Oops - looks like you applied the fix to a subsequent occurrence a few lines down, but didn't change line 35?

Thanks!

Comment (marsupial) [reply pxr/usd/plugin/usdAbc/testenv/testUsdAbcIsConstant.py]: Indeed...corrected

**Pull request merged**